webapp/app/js/perf.js

```diff
@@ -310,6 +336,48 @@ perf.controller('PerfCtrl', [ '$state', '$stateParams', '$scope', '$rootScope',
       plotGraph();
     }
 
+    $scope.resetHighlight = function() {
+      $scope.seriesList.forEach(function(series) {
+        series.highlighted = [];
+      });
+      $scope.highlightedRevision = '';
+      $scope.revisionToHighlight = "";
+      $scope.resetHighlightButton = false;
+      updateURL();
+      plotGraph();
+    }
+
+    $scope.addHighlightedRevision = function() {
+      var rev = $scope.revisionToHighlight;
+      if (rev.length == 12) {
+        $scope.highlightedRevision = rev;
+        $scope.highlightRevision();
+      }
+    }
+
+    $scope.highlightRevision = function() {
+      var rev = $scope.highlightedRevision;
+      if (rev.length == 12) {
+        $q.all($scope.seriesList.map(function(series, i) {
+          if (series.visible) {
+           return $http.get(thServiceDomain + "/api/project/" + series.projectName +
+             "/resultset/?format=json&revision=" + rev + "&with_jobs=false").then(
+            function(response) {
+              if (response.data.results.length > 0) {
+                var result_set_id = response.data.results[0].id;
+                var j = series.flotSeries.resultSetData.indexOf(result_set_id);
+                var seriesToaddHighlight = _.find($scope.seriesList, function(sr) { return sr.signature == series.signature });
+                seriesToaddHighlight.highlighted = [j, rev];
+              }
+              updateURL();
+              plotGraph();
```

Comment:
We should do this when the q.all() promise is finished, rather than every time we download from http.